### ISSUE TITLE
Can't mock a Send Trait

### ISSUE THREAD

**Issue by asomers** [iredelmeier/pseudo#1]
Pseudo cannot mock a trait that is `Send`.  Example:
```rust
    fn send() {
        pub trait A {
            fn foo(&self);
        }

        struct MockA {
            foo: Mock<(), ()>
        }
        impl A for MockA {
            fn foo(&self) {
                self.foo.call(())
            }
        }

        let mock = MockA{foo: Mock::default()};
        let _ = Box::new(mock) as Box<A + Send>;
    }
```
gives the following error:
```
error[E0277]: `(dyn std::ops::Fn(()) + 'static)` cannot be shared between threads safely
   --> src/t_pseudo.rs:431:17
    |
431 |         let _ = Box::new(mock) as Box<A + Send>;
    |                 ^^^^^^^^^^^^^^ `(dyn std::ops::Fn(()) + 'static)` cannot be shared between threads safely                                                   
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn std::ops::Fn(()) + 'static)`
    = note: required because of the requirements on the impl of `std::marker::Sync` for `std::ptr::Unique<(dyn std::ops::Fn(()) + 'static)>`
    = note: required because it appears within the type `std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>`
    = note: required because it appears within the type `std::option::Option<std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>>`
    = note: required because of the requirements on the impl of `std::marker::Sync` for `std::sync::RwLock<std::option::Option<std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>>>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<std::sync::RwLock<std::option::Option<std::boxed::Box<(dyn std::ops::Fn(()) + 'static)>>>>`
    = note: required because it appears within the type `pseudo::Mock<(), ()>`
    = note: required because it appears within the type `<t_pseudo::t::Pseudo as TestSuite>::send::MockA`
    = note: required for the cast to the object type `dyn <t_pseudo::t::Pseudo as TestSuite>::send::A + std::marker::Send`
```

**Comment by iredelmeier** [comment]
Thanks for the issue! This should be fixed in version 0.2.0, which I've just published.

**Comment by asomers** [comment]
Cool!  This makes pseudo the first mocking library with `Send` mocks.  I've updated the mock shootout.  https://asomers.github.io/mock_shootout/
